src/omnicore/omnicore.cpp

```diff
@@ -551,6 +551,38 @@ int mastercore::GetEncodingClass(const CTransaction& tx, int nBlock)
     bool hasOpReturn = false;
     bool hasMoney = false;
 
+    /* Fast Search
```

Comment:
Hm.. did you test this explicitly? It adds extra steps in some cases, so, assuming all transactions were Omni transactions, I'd assume it's slower. But given that this isn't the case, it probably works.